### PR TITLE
Feature/june w2 helena

### PR DESCRIPTION
**Pull request by howdyitshelena**
- Add cancel modal to editor
- Fix arrow size in ObservationDetailView
- Make sure app is always in portrait mode

### CODE REVIEW

File: src/components/Base/CancelModal/CancelModal.js [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+class CancelModal extends React.Component<Props> {
+  render() {
+    const { onCancel, onContinue, visible } = this.props;
+    return (
+      <Modal
+        animation="slide"
+        transparent
+        visible={visible}
+        onRequestClose={() => {}}
```
Comment (czengg) [inline src/components/Base/CancelModal/CancelModal.js]: You can make this an option prop if you're not gonna use it

File: src/components/Base/CancelModal/CancelModal.test.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+// @flow
+import React from 'react';
+import 'react-native';
+import renderer from 'react-test-renderer';
+import CancelModal from './CancelModal';
+import { createObservation } from '../../../mocks/observations';
+import { createCategory } from '../../../mocks/categories';
+
+describe('CancelModal tests', () => {
```
Comment (czengg) [inline src/components/Base/CancelModal/CancelModal.test.js]: yay tests!!! 🎉

File: src/components/Views/ObservationEditor/ObservationEditor.js [hunk 3]
```diff
@@ -439,13 +445,38 @@ class ObservationEditor extends React.Component<
...
+  handleModalCancel = () => {
+    const {
+      clearSelectedObservation,
+      goToMainCameraView,
+      goToMapView,
+      hideCancelModal,
+      observationSource,
+      updateObservation
+    } = this.props;
+    hideCancelModal();
```
Comment (czengg) [inline src/components/Views/ObservationEditor/ObservationEditor.js]: let do a space above just to keep logic and const declarations separate

File: src/components/Views/ObservationEditor/ObservationEditor.js [hunk 4]
```diff
@@ -439,13 +445,38 @@ class ObservationEditor extends React.Component<
...
+      goToMapView();
+    } else {
+      goToMainCameraView();
+    }
+    clearSelectedObservation();
+  };
+
+  handleModalContinue = () => {
+    const { hideCancelModal } = this.props;
+    hideCancelModal();
```
Comment (czengg) [inline src/components/Views/ObservationEditor/ObservationEditor.js]: same comment as above

File: src/components/Views/ObservationEditor/index.js [hunk 5]
```diff
@@ -72,7 +78,9 @@ function mapDispatchToProps(dispatch: Dispatch): DispatchProps {
           routeName: 'MapView'
         })
       ),
-    showSavedModal: () => dispatch(modalShow('saved'))
+    showSavedModal: () => dispatch(modalShow('saved')),
```
Comment (czengg) [inline src/components/Views/ObservationEditor/index.js]: I think you might need to fix ObservationEditor tests to include these props

File: src/components/Views/ObservationEditor/ObservationEditor.js (hunk 3 above)
Comment (howdyitshelena) [reply src/components/Views/ObservationEditor/ObservationEditor.js]: 👍

File: src/components/Views/ObservationEditor/ObservationEditor.js (hunk 4 above)
Comment (howdyitshelena) [reply src/components/Views/ObservationEditor/ObservationEditor.js]: 👍

File: src/components/Views/ObservationEditor/index.js (hunk 5 above)
Comment (howdyitshelena) [reply src/components/Views/ObservationEditor/index.js]: they seem okay!

File: src/components/Base/CancelModal/CancelModal.test.js (hunk 2 above)
Comment (howdyitshelena) [reply src/components/Base/CancelModal/CancelModal.test.js]: 💯

**Pull request merged**